src/main/kotlin/org/rust/ide/annotator/RsHighlightingAnnotator.kt

```diff
@@ -51,17 +51,40 @@ class RsHighlightingAnnotator : AnnotatorBase() {
         }
 
         return when (element.elementType) {
+            DOLLAR -> RsColor.MACRO // TODO: It would be possible to see whether parent is RsMacroBindingGroups or RsMacroExpansionReferenceGroup or RsMacroBinding
             IDENTIFIER, QUOTE_IDENTIFIER, SELF -> highlightIdentifier(element, parent, holder)
             // Although we remap tokens from identifier to keyword, this happens in the
             // parser's pass, so we can't use HighlightingLexer to color these
             in RS_CONTEXTUAL_KEYWORDS -> RsColor.KEYWORD
             FLOAT_LITERAL -> RsColor.NUMBER
 
-            Q -> if (parent is RsTryExpr) {
+            Q -> if (parent is RsMacroExpansionReferenceGroup || parent is RsMacroBindingGroup) { // TODO: and it the last non-whitespace element in parent
+                RsColor.MACRO
+            } else if (parent is RsTryExpr) {
                 RsColor.Q_OPERATOR
             } else {
                 null
             }
+            MUL -> if (parent is RsMacroExpansionReferenceGroup || parent is RsMacroBindingGroup) { // TODO: and it is the last non-whitespace element in parent
```

Comment:
> // TODO: and it is the last non-whitespace element in parent

What do you mean?

Reply:
The `*` (with the meaning that is to be highlighted by this line 68) is only allowed at the end of a group.

For example, we want to highlight the `*` in `($foo:expr)*`, but not in `($foo:expr*2)`. However, I don't think that the latter is allowed anyway, which is why I didn't do the check. It's still nice to know what this `MUL ->` match is supposed to match in the first place, which is why the comment is still there :)

Reply:
Oh, looks like this case is already covered by the grammar itself

You can check [MacroBindingGroup](https://github.com/intellij-rust/intellij-rust/blob/master/src/main/grammars/RustParser.bnf#L1421) and [MacroExpansionReferenceGroup](https://github.com/intellij-rust/intellij-rust/blob/master/src/main/grammars/RustParser.bnf#L1434) rules.
In all other cases, all these elements will have other parent. For example, `MacroPatternContents`

<img width="954" alt="image" src="https://user-images.githubusercontent.com/2539310/172641276-9f1926ff-3350-4fbe-aa0b-5083a8c2a5cd.png">

Reply:
Nice! In that case, I can amend the comment to just state that the grammar only matches it at the end anyway for `RsMacroBindingGroup` and `MacroExpansionReferenceGroup`.